### PR TITLE
Added docstrings to functions in dowload.py

### PR DESCRIPTION
**Pull request by swapsha96**
I could find two TODOs, few functions without documentation and some unfinished functions. (Requires assistance)

### CODE REVIEW

**Comment by nabobalis** [comment]
@swapsha96 I don't want to be seen as the bad guy but can you please try to not open PRs that do not tackle the issue you quoted?

The doc strings I have no issue with and can be added into SunPy without an issue.

**Comment by swapsha96** [comment]
I am sorry, @nabobalis. I don't remember why I quoted that. I should have explained why I referred to the issue. This is an unintentional mistake. I will keep this in mind, and I would be careful with my words from now. Please let me know if I need to do something else with this PR.

**Comment by nabobalis** [comment]
You don't need to be more careful, I realise my comments come off as quite harsh. 

But the original issue you quote is quite an odd one and I would guess that no one has really looked into since it was reported.

File: sunpy/net/download.py [hunk 1]
```diff
@@ -105,9 +105,15 @@ def _default_error_callback(self, e):
         raise e
 
     def wait(self):
+        """
+        Waits for all files to download completely and then return.
```
Comment (nabobalis) [inline sunpy/net/download.py]: I'm not sure that `completely` needs to be stated.

Comment (swapsha96) [reply sunpy/net/download.py]: Fixed it.

**Comment by nabobalis** [comment]
CI fail due to known doctest issues.

**Comment by swapsha96** [comment]
@Cadair @nabobalis Thanks!

**Pull request merged**